indexing-service/src/main/java/io/druid/indexing/overlord/ForkingTaskRunner.java

```diff
@@ -365,6 +365,7 @@ public TaskStatus call()
                                   )
                               );
 
+                              command.add(String.format("-Ddruid.indexer.taskId=%s", task.getId()));
```

Comment:
something like
```java
@Provides
@LazySingleton
public ChatResponseHeaderFilter chatResponseHeaderBuilder(final DataSourceTaskIdHolder taskIdHolder){
   return new JettyResponseHeaderFilterHolder("/druid/worker/v1/chat/*", ImmutableMap.of(ChatHandlerResource.TASK_ID_HEADER, Strings.nullToEmpty(taskIdHolder.getTaskId())));
}
```

If you need better null handling for null taskId then there should be a way to pass a noop filter or something in there